### PR TITLE
Fix integration tests on windows

### LINKED ISSUE #2908: Integration Tests fail on Windows
## Describe the bug
The following integration tests fail on windows:
```
--- FAIL: TestExecute (0.00s)
    agent_test.go:58:
                Error Trace:    C:/Users/lliljenberg/dev/porter/pkg/agent/agent_test.go:58
                                                        C:/Users/lliljenberg/dev/porter/pkg/agent/agent_test.go:17
                Error:          Received unexpected error:
                                no such file or directory '../../bin/porter'
                Test:           TestExecute
```
TestExecute fails because it assumes the porter binary has no .exe extension AND because the repo contains unix symlinks, which cannot be resolved on windows.

```
--- FAIL: TestPorter_FixPermissions (1.47s)
--- FAIL: TestPorter_FixPermissions_NoConfigFile (1.15s)
--- FAIL: TestResolveBundleReference (0.00s)
--- FAIL: TestResolveBundleReference/reference (2.14s)
--- FAIL: TestInstall_stringParam (1.25s)
--- FAIL: TestRegistry (16.80s)
--- FAIL: TestTelemetrySetup (7.52s)
--- FAIL: TestInvokeCustomAction (3.34s)
--- FAIL: TestStepLevelAndBundleLevelOutputs (3.38s)
--- FAIL: TestPull_ContentDigestMissing (3.44s)
--- FAIL: TestInstall_fileParam (3.62s)
--- FAIL: TestInstall_relativePathPorterHome (3.71s)
--- FAIL: TestArchive_StableDigest (3.78s)
--- FAIL: TestSuppressOutput (4.31s)
--- FAIL: TestExecOutputs (2.40s)
--- FAIL: TestInstall_withDockerignore (2.69s)
--- FAIL: TestSensitiveData (2.54s)
--- FAIL: TestDependenciesLifecycle (2.72s)
```
The above tests needs investigation. I know some of them fail because Porter assumes unix file permission bits. 

Update: Many of them fail when a json which contains a windows path (instead of a unix path) with single backslash is unmarshalled. This makes sense since the json spec requires double backslash to indicate a backslash. For example
```
{
"path": "C:\Users\lliljenberg\dev\porter\build\testdata\credentials\ci.json"
}
```
is invalid json, we need to turn it into
```
{
"path": "C:\\Users\\lliljenberg\\dev\\porter\\build\\testdata\\credentials\\ci.json"
}
```

## To Reproduce
1. Run `mage testintegration` on a windows machine

## Expected behavior
All tests should pass

## Version
Copy the output of `porter version` below

porter v1.0.15-24-g513c0a03 (513c0a03)

### PR DESCRIPTION
**Pull request by ludfjig**
# What does this change
Fixes integration tests so they pass on windows . Also fixes a bug where `porter archive` fails on windows due to not being able to create a temp directory, because it tried to create 2 nesting levels at once.

One windows bug remains which is tracked separately, see #2917 

# What issue does it fix
Closes #2908 

_If there is not an existing issue, please make sure we have context on why this change is needed. See our Contributing Guide for [examples of when an existing issue isn't necessary][1]._

[1]: https://getporter.org/src/CONTRIBUTING.md#when-to-open-a-pull-request

# Notes for the reviewer
Please review agent.go. I have removed printing out the filecontents, because on windows there is no way to determine if a file is executable or not. I would like confirmation that this is ok (or not).

# Checklist
- [ ] Did you write tests?
- [ ] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://getporter.org/src/CONTRIBUTORS.md

### CODE REVIEW

File: pkg/porter/archive.go [hunk 1]
```diff
@@ -336,6 +336,9 @@ func (ex *exporter) addImage(base bundle.BaseImage) error {
 // If the name contains a path separator, all path separators will be replaced with "-".
 func (ex *exporter) createArchiveFolder(name string) (string, error) {
 	cleanedPath := strings.ReplaceAll(afero.UnicodeSanitize(name), string(os.PathSeparator), "-")
+	cleanedPath = strings.ReplaceAll(cleanedPath, "/", "-") // this is needed because on windows, if the bundle name is from a reference
+	// e.g. "ghcr.io/getporter/examples/porter-hello:v0.2.0", the name will be "examples/porter-hello",
+	// so we still need to replace the / with -, which the above line doesn't do (on windows)
```
Comment (ludfjig) [inline pkg/porter/archive.go:341]: I'm wondering why this respected the OS's PathSeparator to begin with...?

Comment (ludfjig) [reply pkg/porter/archive.go:341]: I think we can just use '/', no need for checking PathSeparator here. https://github.com/getporter/porter/pull/2154/files

**Review by schristoff — APPROVED** [review]
Awesome job! I can't tell if I kicked off integration or not but I'll set this to merge if integ is happy

**Pull request merged**